src/utils/googleAnalytics.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import React, { Component, useEffect, useState } from 'react'
+import GoogleAnalytics from 'react-ga'
+import { getGoogleAnalyticsTrackingID } from '~/config'
+import { COOKIES_KEY } from '~/logic/cookies/model/cookie'
+import type { CookiesProps } from '~/logic/cookies/model/cookie'
+import { loadFromCookie } from '~/utils/cookies'
+
+const trackingID = getGoogleAnalyticsTrackingID()
+
+if (!trackingID) {
+  console.error('[GoogleAnalytics] - In order to use google analytics you need to add an trackingID')
+} else {
+  GoogleAnalytics.initialize(trackingID)
+}
+
+
+const withTracker = (WrappedComponent, options = {}) => {
+  const [useAnalytics, setUseAnalytics] = useState(false)
+
+  useEffect(() => {
+    async function fetchCookiesFromStorage() {
+      const cookiesState: CookiesProps = await loadFromCookie(COOKIES_KEY)
+      if (cookiesState) {
+        const { acceptedAnalytics } = cookiesState
+        setUseAnalytics(acceptedAnalytics)
+      }
+    }
+    fetchCookiesFromStorage()
+  }, [])
+
+  const trackPage = (page) => {
+    if (!useAnalytics) {
+      return
+    }
+    GoogleAnalytics.set({
+      page,
+      ...options,
+    })
+    GoogleAnalytics.pageview(page)
+  }
+
+  // eslint-disable-next-line
+  const HOC = class extends Component {
```

Comment:
curious if it's possible to rewrite this with useEffect

Reply:
Done